docs/mocks.md

```diff
@@ -0,0 +1,41 @@
+# Mocks
+
+When creating tests, you might need to override existing function to be able to write isolated tests from external behaviour. To accomplish this, you can use mocks. You can also check that a function was called with certain arguments or even a number of times with a spy.
+
+## mock
+> `mock "function" "body"`
+
+Override the behaviour of a function
+
+*Example:*
+```bash
+function test_success() {
+  mock ps echo hello world
+  assert_equals "hello world" "$(ps)"
+}
+```
+
+## spy
+> `spy "function"
+
+Override the behaviour of a function
```

Comment:
What's the digference with a mock?

Reply:
Add links to all spy assertions

Reply:
Those 3 functions are displayed inside the spy docs examples. Should we define them as separate functions? 🤔 
```bash
assert_have_been_called
assert_have_been_called_with
assert_have_been_called_times
```